main.tf

```diff
@@ -1,521 +1,244 @@
+  statement {
+    actions = ["sts:AssumeRole"]
+    principals {
+      type        = "Service"
+      identifiers = ["codebuild.amazonaws.com"]
+    }
+    effect = "Allow"
+  }
+}
+
+# IAM policy for CodeBuild, granting access to S3, CloudWatch Logs, etc.
+data "aws_iam_policy_document" "codebuild_policy" {
+  statement {
+    actions = [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+      "iam:PassRole",
+      "s3:GetObject",
+      "s3:PutObject",
+      "s3:ListBucket",
+      "s3:GetBucketAcl",
+      "s3:GetBucketLocation",
+      "codebuild:StartBuild",
+      "codebuild:BatchGetBuilds",
+      "codebuild:BatchGetProjects"
+    ]
+    resources = [
+      module.s3_bucket.arn,
+      "${module.s3_bucket.arn}/*",
+      "arn:aws:logs:${var.aws_region}:${data.aws_caller_identity.default.account_id}:log-group:/aws/codebuild/${var.project_name}*"
+    ]
+    effect = "Allow"
+  }
+}
+
+# Attach the policy to the IAM role
+resource "aws_iam_role_policy" "codebuild_policy" {
+  name   = "${var.project_name}-codebuild-policy"
+  role   = aws_iam_role.codebuild_role.id
+  policy = data.aws_iam_policy_document.codebuild_policy.json
+}
+
+# S3 bucket for Artifact 
+module "s3_bucket" {
+  source = "github.com/launchbynttdata/tf-aws-module_collection-s3_bucket.git?ref=1.0.0"
+
+  logical_product_family  = var.logical_product_family
+  logical_product_service = var.logical_product_service
+  region                  = var.aws_region
+  class_env               = var.class_env
+
+  block_public_acls       = var.block_public_acls
+  block_public_policy     = var.block_public_policy
+  restrict_public_buckets = var.restrict_public_buckets
+  ignore_public_acls      = var.ignore_public_acls
+
+  kms_s3_key_arn                     = aws_kms_key.kms_key.arn
+  kms_s3_key_sse_algorithm           = var.kms_s3_key_sse_algorithm
+  bucket_key_enabled                 = var.bucket_key_enabled
+  use_default_server_side_encryption = var.use_default_server_side_encryption
+
+  enable_versioning        = var.enable_versioning
+  lifecycle_rule           = var.lifecycle_rule
+  metric_configuration     = var.metric_configuration
+  analytics_configuration  = var.analytics_configuration
+  bucket_name              = local.cache_bucket_name
+  tags                     = var.tags
+  object_ownership         = var.object_ownership
+  control_object_ownership = var.control_object_ownership
+  acl                      = var.acl
+}
+
+# AWS KMS Key Resource
+resource "aws_kms_key" "kms_key" {
+  description             = var.kms_key_description
+  deletion_window_in_days = var.kms_key_deletion_window_in_days
+  enable_key_rotation     = true
+}
+
+resource "random_string" "bucket_prefix" {
+  count   = var.codebuild_enabled ? 1 : 0
+  length  = 12
+  numeric = false
+  upper   = false
+  special = false
+  lower   = true
+}
+
+# Codebuild Project
+resource "aws_codebuild_project" "default" {

```

Comment:
If we're following the pattern of our other modules, this primitive would create only the codebuild project, and then we'd create a collection module with all the other components (artifact bucket, iam roles, etc)

Reply:
I have revised the module, taking away the S3 bucket and other resources. However, Codebuild requires an IAM role, so that's only extra resource in the module.